feat(linter): add support for `extends` property in oxlintrc

camchenry opened the pull request:
- part of https://github.com/oxc-project/oxc/issues/9307

Allows oxlint configuration files to include an `extends` key at the top level. For now (until we decide to change it), this is strictly treated as a list of file paths and we don't support string values like `eslint:recommended` or anything like that.

camchenry commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9217?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9217** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9217?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9217?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9156** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9156?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9155** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9155?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9153** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9153?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9152** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9152?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9217).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camchenry commented:
@Boshen @Sysix @camc314 This stack is ready to be reviewed. I think merging the rest of these PRs should make nested configurations actually useful. It will allow nested configuration files to extend other config files and inherit rules, plugins, and overrides from them.

After we merge all of these, I'll try to write some documentation on the website and we should do some more thorough testing and bug fixing as well.

Sysix reviewed (approved):
Great work 👍 
I added comments in some PRs but they should be all non blocking :)

camchenry commented:
### Merge activity

* **Mar 3, 9:36 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
feat(linter): add support for `extends` property in oxlintrc (#9217)

- part of https://github.com/oxc-project/oxc/issues/9307

Allows oxlint configuration files to include an `extends` key at the top level. For now (until we decide to change it), this is strictly treated as a list of file paths and we don't support string values like `eslint:recommended` or anything like that.
